projects/observability/src/pages/apis/service-detail/service-detail.component.ts

```diff
@@ -35,18 +44,31 @@ export class ServiceDetailComponent {
     }
   ];
 
-  public constructor(private readonly featureStateResolver: FeatureStateResolver) {
-    /**
-     * TODO: Create a generic config to map tabs to feature flags if more tabs
-     * are to be added in the future: https://github.com/razorpay/hypertrace-ui/pull/80#issuecomment-1235141757
-     */
-    this.featureStateResolver.getFeatureState(ApplicationFeature.InstrumentationQuality).subscribe(
-      featureState =>
-        featureState === FeatureState.Enabled &&
-        this.tabs.push({
-          path: 'instrumentation',
-          label: 'Instrumentation Quality'
-        })
-    );
-  }
+  public readonly featureFlaggedTabs: NavigableTab[] = [
+    {
+      path: 'instrumentation',
+      label: 'Instrumentation Quality',
+      flagName: ApplicationFeature.InstrumentationQuality
+    },
+    {
+      path: 'deployments',
+      label: 'Deployments',
+      flagName: ApplicationFeature.DeploymentMarkers
+    }
+  ];
+
+  public enabledTabs$: Observable<NavigableTab[]> = forkJoinSafeEmpty(
+    this.featureFlaggedTabs.map(tab => this.featureStateResolver.getFeatureState(tab.flagName!))
+  ).pipe(
+    map(featureVals => {
+      featureVals.map((featureVal, index) => {
+        if (featureVal === FeatureState.Enabled) {
+          this.tabs.push(this.featureFlaggedTabs[index]);
+        }
+      });
+    }),
+    map(() => this.tabs)
```

Comment:
Do we need two map operators here? Or can we just do `return this.tabs;` in the first one and remove the second one?

Reply:
makes sense, done